### PR TITLE
implemented the update_available function and added test cases for it

### LINKED ISSUE #40: Inventory DB Update - Availability
## As an administrator
**I need** to be able to update an inventory's ```availability``` status
**So that** we can keep track of the stock levels in our inventory 

**Assumptions:**
* The desired Database record exists AND is active

**Acceptance Criteria:**
```
Given the modified availability
When I query the record
Then it mus return with with a 200 if available=1, or 404 if available=0
```

## As a developer
**I need** to create:
1. Functions in ```model.py``` and ```service.py``` to update ```available``` attribute of an existing inventory record in the DB
2. Write Test cases for those functions
3. Corresponding Functions in service.py to setup a resource endpoint

**Assumptions:**
2. model.py is created
3. service.py is created
4. test_model.py is created
5. test_service.py is created

### PR DESCRIPTION
**Pull request by kainattnaeem**

### CODE REVIEW

File: service/service.py [hunk 1]
```diff
@@ -258,6 +258,32 @@ def update_stock(product_id, condition, operation, amount):
...
+    
+
+    if available != 0 or available!=1:
+        return bad_request("Incorrect value for available, can only accept 0 or 1")
+
+    prod = Inventory.find(product_id, condition)
+    if not prod:
+        raise NotFound("The product ID, condition pair does not exist.")
+
+    prod.available = available
```
Comment (arahant) [inline service/service.py:279]: Can you add more logic here?
1. If the ```quantity==0``` and if the request wants to update ```available=1```, then the req is invalid
```
if prod.quantity==0 and available==1:
    raise bad_request("Cannot update available=1, when quantity=0")
else:
    prod.available = available
    prod.update()

File: service/service.py [hunk 2]
```diff
@@ -258,6 +258,32 @@ def update_stock(product_id, condition, operation, amount):
...
+    prod.update()
+
+    if available==0:
```
Comment (arahant) [inline service/service.py]: I think you mean ```if available==1:```

File: service/service.py (hunk 1 above)
Comment (kainattnaeem) [reply service/service.py:279]: done

File: service/service.py (hunk 2 above)
Comment (kainattnaeem) [reply service/service.py]: oops, fixed it

File: service/service.py [hunk 3]
```diff
@@ -258,6 +258,34 @@ def update_stock(product_id, condition, operation, amount):
     app.logger.info(msg_tmp+"%d items having product_id %d and condition %s.", amount, product_id, condition)
     return make_response(jsonify(inventory.serialize()), status.HTTP_200_OK)
 
+################################################################################
+# UPDATE AN EXISTING PRODUCT'S AVAILABILITY
+################################################################################
+@app.route("/inventory/<int:product_id>/<string:condition>/<int:available>", methods=["PUT"])
+def update_stock(product_id, condition, available):
```
Comment (arahant) [inline service/service.py]: @kainattnaeem, be sure to change the name of the function when you replicate it :)

Comment (kainattnaeem) [reply service/service.py]: sorry about that, fixing it

**Pull request merged**